src/apify/_actor.py

```diff
@@ -267,29 +269,31 @@ async def finalize() -> None:
         self._is_initialized = False
 
         if is_running_in_ipython():
-            self.log.debug(f'Not calling sys.exit({exit_code}) because Actor is running in IPython')
+            self.log.debug(f'Not calling sys_exit({exit_code}) because Actor is running in IPython')
         elif os.getenv('PYTEST_CURRENT_TEST', default=False):  # noqa: PLW1508
-            self.log.debug(f'Not calling sys.exit({exit_code}) because Actor is running in an unit test')
+            self.log.debug(f'Not calling sys_exit({exit_code}) because Actor is running in an unit test')
         elif hasattr(asyncio, '_nest_patched'):
-            self.log.debug(f'Not calling sys.exit({exit_code}) because Actor is running in a nested event loop')
+            self.log.debug(f'Not calling sys_exit({exit_code}) because Actor is running in a nested event loop')
         else:
-            sys.exit(exit_code)
+            sys_exit(exit_code)
```

Comment:
I believe this logic could be simplified a fair bit. 

- we don't need to check if we're running under pytest - we can instead pass in a mock `sys_exit`
- didn't @vdusek manage to remove the nesting?
- we probably still have to check for ipython :confused:

Reply:
Regarding the `nest-asyncio` ...

We managed to resolve the problem of executing the Actor & Spider together. Here is more context and the core result structure - https://github.com/scrapy/scrapy/discussions/6639#discussioncomment-12019270.

However, there's still one remaining challenge - executing asyncio coroutines from the synchronous Scheduler (and HTTP cache storage) code. This is in-progress. 

Once that's addressed, I'll open a new PR removing `nest-asyncio` completely.